contents/javascript/Javascript-framework-React.md

```diff
@@ -0,0 +1,134 @@
+<frontmatter>
+  title: React
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# React
+
+Author: [Aadyaa Maddi](https://github.com/amad-person)
+
+### What is React?
+
+**React** is an open-source JavaScript <tooltip content="React only handles the view component of your application. It is not an MVC framework!">library</tooltip> for building composable user interfaces. It was developed by Facebook and released to the world in 2013, and is now used in a huge number of popular applications, including Facebook and Instagram.
+
+<box type="tip">
+    If you have the <a href="https://github.com/facebook/react-devtools">React DevTools</a> extension installed in your browser, you can inspect the webpage of either of these applications and observe that a lot of React components are being used by them! 
+</box>
+
+React abstracts away the [DOM](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Introduction) from you, giving you a simpler programming model and better performance. It can also render on the server using Node, and it can power native mobile applications using [React Native](https://facebook.github.io/react-native/).
+
+### Why React?
+
+React allows developers to create large web applications which can change data, without reloading the page. The main purpose of React is to be fast, scalable, and simple. 
+Some of its benefits are described in detail below.
+
+#### React is Declarative
+
+With React's <tooltip content="Declarative programming focuses on what the program should accomplish without specifying how the program should achieve the result.">declarative</tooltip> approach to building user interfaces, you can painlessly create interaction-rich applications. 
+You can build these interfaces without touching the DOM directly, and even have an event system that doesn't need you to interact with actual DOM events. 
```

Comment:
without ~touching~ **manipulating** the DOM directly, and even have an event system that doesn't ~need~ **require** you to?